src/execution/operator/helper/physical_limit.cpp

```diff
@@ -62,49 +43,36 @@ SinkResultType PhysicalLimit::Sink(ExecutionContext &context, GlobalSinkState &g
 
 	// get the next chunk from the child
 	if (limit == INVALID_INDEX) {
-		limit = GetDelimiter(input, limit_expression.get(), 1ULL << 62ULL);
+		limit = 1ULL << 62ULL;
+		Value val = GetDelimiter(input, limit_expression.get());
+		if (!val.is_null) {
+			limit = val.GetValue<idx_t>();
+		}
+		if (limit > 1ULL << 62ULL) {
+			throw BinderException("Max value %lld for LIMIT/OFFSET is %lld", limit, 1ULL << 62ULL);
+		}
 	}
 	if (offset == INVALID_INDEX) {
-		offset = GetDelimiter(input, offset_expression.get(), 0);
+		offset = 0;
+		Value val = GetDelimiter(input, offset_expression.get());
+		if (!val.is_null) {
+			offset = val.GetValue<idx_t>();
+		}
+		if (offset > 1ULL << 62ULL) {
+			throw BinderException("Max value %lld for LIMIT/OFFSET is %lld", offset, 1ULL << 62ULL);
+		}
 	}
 	idx_t max_element = limit + offset;
-	idx_t input_size = input.size();
+	if (limit == INVALID_INDEX) {
+		max_element = INVALID_INDEX;
```

Comment:
This line does not appear to be covered. When should this scenario occur?

Reply:
Your are right, This part is not necessary. I will remove it.